utils/http/errors.py

```diff
@@ -0,0 +1,34 @@
+# Django Imports
+from django.http import HttpResponse
+
+
+class HttpErrorResponseError(Exception):
+    """Generic Response Error exception
+
+    It helps to stop processing a view function by raising an exception.
+    It is like `Http404` exception but with ability to define your response.
+
+    Status code defaults to 404 but can be overridden if response is string.
+    Status code is not changed if response is derived from `HttpResponse`.
+
+    NOTE: `htk.middleware.classes.HttpErrorResponseMiddleware` MUST be in
+    MIDDLEWARES in Django Settings.
+    """
+
+    def __init__(self, response, status_code=None):
+        if isinstance(response, str):
+            self.response = HttpResponse(response)
+            self.response.status_code = (
+                status_code if status_code is not None else 404
+            )
```

Comment:
```suggestion
    def __init__(self, response, status_code=400):
        if isinstance(response, str):
            self.response = HttpResponse(response)
            self.response.status_code = status_code
```

- Why not use a default arg?
- Prefer `400` over `404`

Actually, can are you able to set the status code at the time of building, something like `HttpResponse(response, status_code=status_code)` ?

Reply:
- Used default arg
- Used 400 instead of 404 as default

Yes, the arg name is `status` and used it.